server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -243,39 +306,63 @@ public String executor() {
             }
         };
         try {
-            transportService.connectToExtensionNode(extensionNode);
+            logger.info("Sending extension request type: " + REQUEST_EXTENSION_ACTION_NAME);
+            transportService.connectToExtensionNode(extension);
             transportService.sendRequest(
-                extensionNode,
+                extension,
                 REQUEST_EXTENSION_ACTION_NAME,
-                new PluginRequest(transportService.getLocalNode(), new ArrayList<DiscoveryExtensionNode>(uninitializedExtensions)),
-                pluginResponseHandler
+                new InitializeExtensionRequest(transportService.getLocalNode(), extension),
+                initializeExtensionResponseHandler
             );
+            // TODO: make asynchronous
```

Comment:
Sorry, I remember we discussed that before, this `TODO: make asynchronous` for this pull request or there is issue / pull request to follow up?

Reply:
@reta right here: https://github.com/opensearch-project/OpenSearch/issues/5595